### PR TITLE
Separate Endpoints into a standalone deployment

### PR DESCRIPTION
**Pull request by nb-ohad**
- Remove endpoint deployment and reconcile for NooBaa core sts
- Open mongo to external connections
- Create an endpoint deployment yaml
- Add endpoints control and status into noobaa CRD
- Change the way we handle resource limit/requets for the NooBaa pods
  * Replace all default values with values decided for base profile.
  * Expose endpoint pod resource configuration inside the endpoint part of the NooBaa crd
  * Changed test-olm to use a low profile for endpoint resources
- Fix test to support new endpoint configuration

### CODE REVIEW

File: pkg/system/phase4_configuring.go [hunk 1]
```diff
@@ -51,10 +57,110 @@ func (r *Reconciler) ReconcilePhaseConfiguring() error {
...
+			for j := range c.Env {
+				switch c.Env[j].Name {
+				case "MGMT_URL":
+					c.Env[j].Value = fmt.Sprintf(`https://%s.%s.svc`,
+						r.ServiceMgmt.Name, r.Request.Namespace)
+
+				case "MONGODB_URL":
+					// TODO: change to mongodb pod name after mongo separation
+					c.Env[j].Value = fmt.Sprintf(`mongodb://%s-0.%s/nbcore`,
+						r.CoreApp.Name, r.ServiceMgmt.Name)
```
Comment (dannyzaken) [inline pkg/system/phase4_configuring.go]: I think the pod dns name should be (r.NooBaaDB.Name)-0.(r.NooBaaDB.Spec.ServiceName)

Comment (nb-ohad) [reply pkg/system/phase4_configuring.go]: Changed the code as of your suggestion

File: test/cli/test_cli_functions.sh [hunk 2]
```diff
@@ -100,14 +100,16 @@ function timeout {
 
 function install {
 # TODO: once we can control the core resource amount we can use it,
-#       for now we will mimic the install command 
+#       for now we will mimic the install command
     noobaa crd create
     noobaa system create
```
Comment (guymguym) [inline test/cli/test_cli_functions.sh]: We already support --core-resources and --db-resources and --endpoint-resources as flags on `noobaa system create` so we can remove those kubectl patches

Comment (guymguym) [reply test/cli/test_cli_functions.sh]: @nb-ohad just like you did for test-olm.sh

Comment (nb-ohad) [reply test/cli/test_cli_functions.sh]: Changed the test to use the new --mini flag in #183

**Pull request merged**